src/prommis/conn/connectivity.py

```diff
@@ -0,0 +1,581 @@
+"""
+Calculate and display model connectivity.
+
+The main thing this module does is create and process a connectivity
+matrix encoded as follows:
+
+    Units,Unit 1 name,Unit 2 Name, ..., Unit N Name
+    Arcs,,,,,,,	...,
+    Arc1 Name,	-1,	0, 0, ..., 0
+    Arc2 Name, 0 , 1, 0 ,... , 0
+
+Where each cell at the intersection of an Arc (row i) and Unit (column j)
+is either:
+  *  -1 meaning Arc(i) is an outlet of Unit(j), 
+  *  1 meaning Arc(i) is an inlet for Unit(j),
+  *  0 meaning there is no connection
+
+
+This module can be run as a script or used programmatically, using the
+public functions `create_from_matrix` and `create_from_model`.
+"""
+
+import argparse
+import csv
+from dataclasses import dataclass, field
+import enum
+import importlib
+from io import StringIO
+import logging
+from pathlib import Path
+import pprint
+import sys
+from tempfile import TemporaryFile
+from traceback import format_stack
+from typing import TextIO
+import warnings
+
+try:
+    import pyomo
+    from pyomo.network import Arc
+except ImportError as err:
+    pyomo = None
+    warnings.warn(f"Could not import pyomo: {err}")
+
+# For logging
+from prommis.conn.util.yoder import add_log_options, process_log_options
+
+# Constants
+AS_STRING = "-"
+
+# Logging
+# This variable is reassigned if run as script
+_log = logging.getLogger(__name__)
+
+
+##############
+# Classes
+##############
+
+
+class OutputFormats(enum.Enum):
+    markdown = "markdown"
+    html = "html"
+    mermaid = "mermaid"
+    csv = "csv"

```

Comment:
To resolve the Pylint errors for these class attributes, you can add the following to `pyproject.toml` after 

```
class-const-naming-style = "any"
```

More info: https://pylint.readthedocs.io/en/v2.17.7/user_guide/messages/convention/invalid-name.html#predefined-naming-styles

Reply:
I decided all-caps was OK for them anyways, but thanks for the pointer